### ISSUE TITLE
[REQUEST] Support for multiple swagger URLS with different enpoints.

### ISSUE THREAD

**Issue by hasannadeem** [rswag/rswag#761]
## Is your feature request related to a problem? Please describe.
My app has two users one is seller and second one is admin, I have seller apis and admin apis, The admin apis should be admin authenticated and seller apis are public for the seller.

Currently, i have one swagger doc for the admin i.e `admin/api-docs/index.html` that is admin authorized (only admins can access it), i wanted to create another publich document dor sellers with url `seller/api-docs/index.html` that will totally different endpoints. I searched the documentation and can't seem to find anything.

Is there any way to do it?

I also don't know how it would work with rake rswag:specs:swaggerize i suppose there would be seperate .yml files would this command update both?

**Comment by kevinjcoleman** [comment]
@hasannadeem did you ever figure this out?

**Comment by gap777** [comment]
@hasannadeem Im interested in your solution as well

**Comment by teemutammela** [comment]
Here's how I got it to work. My application also has different namespaces for internal API and public API, each with their corresponding Swagger YAML files.

**initializers/rswag_ui.rb**
```ruby
Rswag::Ui.configure do |c|
  c.openapi_endpoint "/docs/api/v1/swagger.yaml", "Internal API V1 Docs"
  c.openapi_endpoint "/docs/public_api/v1/swagger.yaml", "Public API V1 Docs"
end
```

**config/routes.rb**
```ruby
Rails.application.routes.draw do
  mount Rswag::Ui::Engine => "/docs"
  mount Rswag::Api::Engine => "/docs"

  # Rest of the routes
end
```

**spec/swagger_helper.rb**
```
config.openapi_specs = {
  "api/v1/swagger.yaml" => {
      # Internal API config
    },
    "public_api/v1/swagger.yaml" => {
       # Public API config
    }
}
```

Make sure you have the `openapi_spec` parameter in your tests so they are listed only in the appropriate file:

```ruby
RSpec.describe "YourModel", openapi_spec: "public_api/v1/swagger.yaml", type: :request do
  # Rest of the thest
end
```

**Comment by kevinjcoleman** [comment]
@teemutammela does that handle authenticating differently for internal vs external?

**Comment by teemutammela** [comment]
@kevinjcoleman Yes it does. My app has separate controllers for internal and public API. The default `ApplicationController` for the internal API and `PublicApiController` for the public API. Each have their own authentication methods, which can be described in `swagger_helper.rb`.

**Comment by kevinjcoleman** [comment]
@teemutammela I meant separate authentication for the docs themselves, the OpenAPI spec config being different makes sense.

**Comment by teemutammela** [comment]
@kevinjcoleman I'm not using authentication for the docs, so I haven't tried if it's possible to set different username and password for each `openapi_endpoint`.
